### ISSUE TITLE
Selection does not consider line-spacing

### ISSUE THREAD

**Issue by albertmink** [eclipse-platform/eclipse.platform.ui#2307]
Let's make sure issue is not already fixed in latest builds first.

- [x] I verified I can reproduce this issue against [latest Integration Build of Eclipse SDK](https://download.eclipse.org/eclipse/downloads/)

## Steps to reproduce

From a fresh installation and clean workspace:

* *Set line spacing > 0*
<img width="697" alt="image" src="https://github.com/user-attachments/assets/77a6941e-2a92-4d2c-aa0e-984cd3cef115">

I tried

* *select a character or a word *

I expected: *that selection marker extend over the line-spacing*

But got: *The selected character/word is displayed as if line-spacing were 0*
<img width="325" alt="select-with-line-spacing-40" src="https://github.com/user-attachments/assets/aa911e16-75fa-40e7-a624-d3574b841fdc">

## Here is some relevant log output

From `<workspace>/.metadata/.log`
```
<relevant log output>
```

## Tested under this environment:
* OS & version: ...
* Eclipse IDE/Platform version (as shown in *Help > About*): ...
* ...Anything else that can be useful...

## Community

- [x] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

**Comment by jukzi** [comment]
Did you read that?
![image](https://github.com/user-attachments/assets/720d9d6c-d08c-40ab-b269-414e5baa6767)
Do you plan to contribute a PR that does the job?

**Comment by BeckerWdf** [comment]
> Did you read that? ![image](https://private-user-images.githubusercontent.com/51790620/369409342-720d9d6c-d08c-40ab-b269-414e5baa6767.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4MzcyMjksIm5iZiI6MTcyNjgzNjkyOSwicGF0aCI6Ii81MTc5MDYyMC8zNjk0MDkzNDItNzIwZDlkNmMtZDA4Yy00MGFiLWIyNjktNDE0ZTViYWE2NzY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDEyNTUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwZGYxYWY2ZDY0ZmFmZGFlNjk4Y2MyMDJmYWNiMzZjNGY1NWU0MTVmNmJkMDRhMTMxZTE3ZGM4ZDc5ZDVlY2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hie-flxxxZttXvTQIr9pajHCoohWw802X1JYOQbcFjs) Do you plan to contribute a PR that does the job?

I don't think that his sentence relates to this issue. The extra linespacing setting has some issues - so I would say it's simply a bug.
Pls. have a look at https://github.com/eclipse-platform/eclipse.platform.swt/pull/1329.
There we did a fix for the selection in such cases. How is your case different?

**Comment by albertmink** [comment]
@BeckerWdf let's consider three cases: ( I just wanted to play with this feature 🥲 )

(1) see initial post.
<img width="325" alt="select-with-line-spacing-40" src="https://github.com/user-attachments/assets/6afb66c8-61f8-4dc3-b7c3-2d3be7c874d2">

(2) selecting multi line (partial) 
<img width="310" alt="Screenshot 2024-09-20 at 15 49 32" src="https://github.com/user-attachments/assets/116f688c-22b9-4085-950b-dab64f2f0708">
(3) selecting multi line (entire)
<img width="295" alt="Screenshot 2024-09-20 at 15 50 03" src="https://github.com/user-attachments/assets/a4daf842-edbb-4240-828b-8968a4a78869">

Let's say only (3) looks good.
